samplers/jaegerremote/sampler_remote_test.go

```diff
@@ -586,3 +587,16 @@ func TestSamplingStrategyParserImpl_Error(t *testing.T) {
 	require.Error(t, err, "output: %+v", val)
 	require.Contains(t, err.Error(), `unknown value "foo_bar"`)
 }
+
+func TestDefaultSamplingStrategyFetcher_Panic(t *testing.T) {
+	defaultTransport := http.DefaultTransport
+	http.DefaultTransport = http.NewFileTransport(http.Dir("/"))
+
+	t.Cleanup(func() {
+		http.DefaultTransport = defaultTransport
+	})
+
+	require.Panics(t, func() {
```

Comment:
I don't think this test should be added, it tests an internal implementations detail as if that is a requirement for default fetcher to fail on a different transport.

Reply:
The opposite test makes more sense, that the sampler does not fail when used with a custom fetcher and the default transport is overwritten.

Reply:
> the sampler does not fail when used with a custom fetcher

Whether a sampler fails or not depends on its implementation. 

The test I added exposes an issue with the current implementation, so we can all agree that `WithSamplingStrategyFetcher` should be public to at least be able to avoid the issue.

Reply:
> so we can all agree that `WithSamplingStrategyFetcher` should be public to at least be able to avoid the issue.

Yes and the way to demonstrate that is with a test that does NOT fail when used with custom sampler. The fact that the existing sampler panics is completely irrelevant (and is frankly a bug), not something that should be codified as expected behavior with a test.

Reply:
@morus12 I think it is a bug as well.

1. I suggest to create an issue for it. There should be on panic.
2. You can still have the test but you should add a comment that this test demonstrates a bug which should be fixed and reference the issue created in point 1.